test/framework/integration_framework/integration_test_framework.hpp

```diff
@@ -176,9 +180,8 @@ namespace integration_framework {
       ObjectType &ref_for_insertion,
       const WaitTime &wait,
       const std::string &error_reason) {
-    if (!queue.try_pop(ref_for_insertion)) {
-      std::this_thread::sleep_for(wait);
-    }
+    std::unique_lock<std::mutex> lk(queue_mu);
+    queue_cond.wait_for(lk, wait, [&]() { return not queue.empty(); });
```

Comment:
What's the point of the unused predicate? Maybe assert is missing
also `()` are redundant

Reply:
@l4l wat?